Actions/DetermineProjectsToBuild/DetermineProjectsToBuild.psm1

```diff
@@ -76,32 +87,91 @@ function Get-FilteredProjectsToBuild($settings, $projects, $baseFolder, $modifie
+        $project,
+        [Parameter(HelpMessage = "The base folder", Mandatory = $true)]
+        $baseFolder,
+        [Parameter(HelpMessage = "A list of modified files", Mandatory = $true)]
+        $modifiedFiles
+    )
+    Write-Host "Determining whether to build project $project based on modified files"
+
+    $projectFolders = GetProjectFolders -baseFolder $baseFolder -project $project -includeAlGoFolder
+
+    #Include the base folder in the modified files
+    $modifiedFiles = @($modifiedFiles | ForEach-Object { return Join-Path $baseFolder $_ })
+
+    $modifiedProjectFolders = @($projectFolders | Where-Object {
+        $projectFolder = Join-Path $baseFolder "$_/*"
+
+        return $($modifiedFiles -like $projectFolder)
+    })
+
+    if ($modifiedProjectFolders.Count -gt 0) {
+        Write-Host "Modified files found for project $project : $($modifiedProjectFolders -join ', ')"
+        return $true
+    }
+
+    Write-Host "No modified files found for project $project. Not building project"
+    return $false
+}
 
-    $filteredProjects = @()
-    foreach($project in $projects)
-    {
-        if (Test-Path -Path (Join-Path $baseFolder "$project/.AL-Go/settings.json")) {
-            $projectFolders = GetProjectFolders -baseFolder $baseFolder -project $project -includeAlGoFolder
+<#
+.Synopsis
+    Creates buils dimensions for a list of projects.
 
-            $modifiedProjectFolders = @($projectFolders | Where-Object {
-                $projectFolder = Join-Path $baseFolder "$_/*"
+.Outputs
+    An array of build dimensions for the projects and their corresponding build modes.
+    Each build dimension is a hashtable with the following keys:
+    - project: The name of the AL-Go project
+    - buildMode: The build mode to use for the project
+#>
+function CreateBuildDimensions {
+    param(
+        [Parameter(HelpMessage = "A list of AL-Go projects for which to generate build dimensions")]
+        $projects = @(),
+        $baseFolder
+    )
 
-                return $($modifiedFiles -like $projectFolder)
-            })
+    $buildDimensions = @()
 
-            if ($modifiedProjectFolders.Count -gt 0) {
-                # The project has been modified, add it to the list of projects to build
-                $filteredProjects += $project
+    foreach($project in $projects) {
+        $projectSettings = ReadSettings -project $project -baseFolder $baseFolder
+        $buildModes = @($projectSettings.buildModes)
+
+        if(!$buildModes) {
+            Write-Host "No build modes found for project $project, using default build mode 'Default'."
+            $buildModes = @('Default')
+        }
+
+        foreach($buildMode in $buildModes) {
+            $buildDimensions += @{
+                project = $project
+                projectName = $projectSettings.projectName
+                buildMode = $buildMode
             }
         }
     }
 
-    return $filteredProjects
+    return @(, $buildDimensions) # force array
```

Comment:
We don't really use this hack anywhere else, and I don't like it - it adds an empty item to the array and you cannot use count etc. typically we cast the function as an array.